draft-peabody-dispatch-new-uuid-format-03.txt

```diff
@@ -528,1005 +546,889 @@ Internet-Draft               new-uuid-format               February 2022
-   is that timestamp values generated by one system and parsed by
-   another are guaranteed to have sub-second precision of either the
-   generator or the parser, whichever is less.  Additionally, the system
-   parsing the UUIDv7 value does not need to know which precision was
-   used during encoding in order to function correctly.
+   UUID version 7 features a time-ordered value field derived from the
+   widely implemented and well known Unix Epoch timestamp source, as
+   well as improved entropy characteristics over versions 1 or 6.
+   Implementations SHOULD utilize UUID version 7 over UUID version 1 and
+   6 if possible.
 
-   The format for the 16-octet, 128 bit UUIDv7 is shown in Figure 2
 
-        0                   1                   2                   3
-        0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-       |                            unixts                             |
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-       |unixts |       subsec_a        |  ver  |       subsec_b        |
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-       |var|                   subsec_seq_node                         |
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-       |                       subsec_seq_node                         |
-       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-
-                   Figure 2: UUIDv7 Field and Bit Layout
 
 
-
-
-Peabody & Davis           Expires 5 August 2022                [Page 10]
+Peabody & Davis          Expires 28 August 2022                [Page 10]
 
 Internet-Draft               new-uuid-format               February 2022
 
 
-   unixts:
-      36 bit big-endian unsigned Unix Timestamp value
-
-   subsec_a:
-      12 bits allocated to sub-second precision values.
+    0                   1                   2                   3
+    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                           unix_ts_ms                          |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |      unix_ts_ms       |                  rand_a               |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |    var_ver    |            rand_b                             |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                            rand_b                             |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 
-   ver:
-      The 4 bit UUIDv7 version (0111)
+                   Figure 4: UUIDv7 Field and Bit Layout
 
-   subsec_b:
-      12 bits allocated to sub-second precision values.
+   unix_ts_ms:
+      44 bit big-endian unsigned number of unix epoch timestamp as per
```

Comment:
is '44' a typo?

Reply:
Silly me, missed them.

This is why we peer review!